web/src/buddy_data.js

```diff
@@ -49,8 +51,8 @@ export function get_user_circle_class(user_id) {
 }
 
 export function level(user_id) {
-    if (people.is_my_user_id(user_id)) {
-        // Always put current user at the top.
+    // Put current user at the top, unless we're in a user search view.
+    if (people.is_my_user_id(user_id) && $(".user-list-filter").val() === "") {
```

Comment:
How do we feel about using UI/jquery to test this logic? I'd prefer some sort of "pass in a variable that says if we're sorting for search or not", but I think that would involve several layers of passing variables around which I don't like.

Reply:
Definitely not desired; in part because this whole module is explicitly a data module that doesn't have any DOM manipulation in it.

I think it's fine to pass data around through a few layers; though I think a global variable for the module's state could also be an appropriate choice?

Reply:
It feels important that we switch to not having the user at the top until there's search text in the search bar, otherwise there'll be a visual jump that happens just by opening the search bar. There's almost always visual change when entering text into the search input field, so that seems like the time to change the sorting.

https://github.com/zulip/zulip/assets/5634097/c7318c98-2c54-4142-8581-f43da61785b4

Reply:
I'm pushing a change that keeps track of this in UserSearch's `this.$input.on("input"` instead -- let me know what you think of that!

Reply:
tests are failing because of test coverage, but before I get to that can you let me know if you like this implementation @timabbott ?